### PR TITLE
Color picker

### PR DESCRIPTION
**Pull request by Lavcodes**
**Description**
#4076 Adds a Color Picker feature to allow user to choose the color they want to highlight the text with.

**Issue**
Fixes: #4076

**Example**
Overview

OverView

How it opens up

OpensUp

Highlight Color Changes

HighlightColorChanges

Different Color

DiffColor

**Context**
Adds an additional feature as earlier the highlight color was quite subtle. This allows User to flexibly choose a color of their choice. Usage can be extended into other features, that people making use of slate may want to use in their own projects.

**Checks**
- [ YES ] The new code matches the existing patterns and styles.
- [ YES ] The tests pass with `yarn test`.
- [ YES ] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [ YES ] The relevant examples still work. (Run examples with `yarn start`.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,7 +106,6 @@
     "slate-hyperscript": "*",
     "slate-react": "*",
     "source-map-loader": "^0.2.4",
-    "typescript": "3.9.7"
```
Comment (dylans) [inline package.json:108]: somehow my attempt to rebase this removed TS which is not what was intended.

File: .github/ISSUE_TEMPLATE/bug-core.md [hunk 2]
```diff
@@ -1,13 +1,11 @@
 ---
+
```
Comment (dylans) [inline .github/ISSUE_TEMPLATE/bug-core.md:2]: Could you please update this PR to not be applying formatting changes to the .github templates?

**Review by dylans — CHANGES_REQUESTED** [review]
Overall this would be a nice example to add. Please see my comments... main thing to change is that this really should not be modifying the issue creation templates, etc. as part of this PR.

Also, Plate recently added a font/font-ui plugin into  https://github.com/udecode/plate/tree/main/packages/marks ... if you have interest, your example has a much better UX and I'm sure it would be a welcome contribution there as well.

File: package.json [hunk 3]
```diff
@@ -83,6 +83,7 @@
     "prettier": "^1.19.1",
     "prismjs": "^1.5.1",
     "react": "^16.12.0",
+    "react-pick-color": "^1.0.1",
     "react-dom": "^16.12.0",
```
Comment (echarles) [inline package.json:87]: I don't see the `react-pick-color` being used in the code. I may miss something looking at the diff?

File: package.json [hunk 4]
```diff
@@ -83,6 +83,7 @@
     "prettier": "^1.19.1",
     "prismjs": "^1.5.1",
     "react": "^16.12.0",
+    "react-pick-color": "^1.0.1",
```
Comment (echarles) [inline package.json:86]: I don't see the react-pick-color being used in the code. I may miss something looking at the diff?

**Comment by dylans** [comment]
I'm closing this one out due to inactivity and the unaddressed feedback. Feel free to either re-open or raise a new PR if you're interested in finishing this addition to the examples. Thanks!

**Pull request closed without merging**